clang/test/APINotes/objcxx-swift-name.m

```diff
@@ -1,22 +1,40 @@
+// RUN: %clang_cc1 -fmodules -fblocks -fimplicit-module-maps -fmodules-cache-path=%t/ModulesCache/CxxInterop -fdisable-module-hash -fapinotes-modules -fsyntax-only -I %S/Inputs/Headers -F %S/Inputs/Frameworks %s -ast-dump -ast-dump-filter gloablFnInNS -x objective-c++ | FileCheck -check-prefix=CHECK-FN %s
+// RUN: %clang_cc1 -fmodules -fblocks -fimplicit-module-maps -fmodules-cache-path=%t/ModulesCache/CxxInterop -fdisable-module-hash -fapinotes-modules -fsyntax-only -I %S/Inputs/Headers -F %S/Inputs/Frameworks %s -ast-dump -ast-dump-filter ImportAsReference -x objective-c++ | FileCheck -check-prefix=CHECK-REF %s
 // RUN: %clang_cc1 -fmodules -fblocks -fimplicit-module-maps -fmodules-cache-path=%t/ModulesCache/CxxInterop -fdisable-module-hash -fapinotes-modules -fsyntax-only -I %S/Inputs/Headers -F %S/Inputs/Frameworks %s -ast-dump -ast-dump-filter "(anonymous)" -x objective-c++ | FileCheck -check-prefix=CHECK-ANONYMOUS-ENUM %s
 
 #import <CXXInteropKit/CXXInteropKit.h>
 
 // CHECK: Dumping NSSomeClass:
-// CHECK-NEXT: ObjCInterfaceDecl {{.+}} imported in CXXInteropKit <undeserialized declarations> NSSomeClass
-// CHECK-NEXT: SwiftNameAttr {{.+}} <<invalid sloc>> "SomeClass"
+// CHECK: ObjCInterfaceDecl {{.+}} imported in CxxInteropKit <undeserialized declarations> NSSomeClass
+// CHECK: SwiftNameAttr {{.+}} <<invalid sloc>> "SomeClass"
 
 // CHECK: Dumping NSSomeClass::didMoveToParentViewController::
-// CHECK-NEXT: ObjCMethodDecl {{.+}} imported in CXXInteropKit - didMoveToParentViewController: 'void'
-// CHECK-NEXT: ParmVarDecl
-// CHECK-NEXT: SwiftNameAttr {{.+}} <<invalid sloc>> "didMove(toParent:)"
+// CHECK: ObjCMethodDecl {{.+}} imported in CxxInteropKit - didMoveToParentViewController: 'void'
+// CHECK: SwiftNameAttr {{.+}} <<invalid sloc>> "didMove(toParent:)"
 
 // CHECK: Dumping SomeClassRed:
-// CHECK-NEXT: EnumConstantDecl {{.+}} imported in CXXInteropKit SomeClassRed 'ColorEnum'
-// CHECK-NEXT: SwiftNameAttr {{.+}} <<invalid sloc>> "red"
+// CHECK: EnumConstantDecl {{.+}} imported in CxxInteropKit SomeClassRed 'ColorEnum'
+// CHECK: SwiftNameAttr {{.+}} <<invalid sloc>> "red"
+
+// CHECK-METHOD: Dumping GlobalStruct::method:
+// CHECK-METHOD: SwiftAttrAttr {{.+}} <<invalid sloc>> Implicit "import_unsafe"
+// CHECK-METHOD: SwiftNameAttr {{.+}} <<invalid sloc>> "globalMethod()"
+
+// CHECK-METHOD: Dumping ParentNS::ChildNS::ParentStruct::ChildStruct::method:
+// CHECK-METHOD: SwiftNameAttr {{.+}} <<invalid sloc>> "nestedMethod()"
+
+// CHECK-FN: Dumping ParentNS::ChildNS::gloablFnInNS:
+// CHECK-FN: SwiftAttrAttr {{.+}} <<invalid sloc>> Implicit "import_unsafe"
+// CHECK-FN: SwiftNameAttr {{.+}} <<invalid sloc>> "childFnInNS()"
+
+// CHECK-REF: Dumping ImportAsReference:
+// CHECK-REF: SwiftAttrAttr {{.+}} <<invalid sloc>> Implicit "import_reference"
```

Comment:
Nit: It would be better if you could have a separate test case file for the new import_as , retain and release attributes.

Reply:
It's a fairly small test file, I'd like to keep it all together as it's testing roughly the same thing.

Reply:
If you feel strongly, I will fix it post-commit.